tests/test_insert_deletes_consolidate.cpp

```diff
@@ -79,59 +75,115 @@ inline void load_aligned_bin_part(const std::string& bin_file, T* data,
 }
 
 std::string get_save_filename(const std::string& save_path,
-                              size_t points_to_skip, size_t points_deleted,
+                              size_t points_to_skip, float points_deleted,
+                              float  points_inserted,
                               size_t last_point_threshold) {
   std::string final_path = save_path;
   if (points_to_skip > 0) {
     final_path += "skip" + std::to_string(points_to_skip) + "-";
   }
-  if (points_deleted > 0) {
-    final_path += "del" + std::to_string(points_deleted) + "-";
+  if (points_deleted > 0 && points_inserted > 0) {
+    final_path += "insert" + std::to_string(points_inserted) + "-" + "delete" +
+                  std::to_string(points_deleted);
+  } else if (points_deleted == 0 && points_inserted > 0) {
+    final_path += "insert" + std::to_string(points_inserted);
+  } else if (points_deleted > 0 && points_inserted == 0) {
+    final_path += "delete" + std::to_string(points_deleted);
   }
-  final_path += std::to_string(last_point_threshold);
   return final_path;
 }
 
 template<typename T, typename TagT>
-void insert_till_next_checkpoint(diskann::Index<T, TagT>& index, size_t start,
-                                 size_t end, size_t thread_count, T* data,
-                                 size_t aligned_dim) {
+void insert_till_next_checkpoint(diskann::Index<T, TagT>& index,
+                                 size_t max_points, float insert_percentage,
+                                 float build_percentage, size_t thread_count,
+                                 T* data, size_t aligned_dim,
+                                 float                delete_percentage,
+                                 diskann::Parameters& delete_params) {
   diskann::Timer insert_timer;
+  int64_t        insert_points = max_points * insert_percentage;
+  int64_t        delete_points = max_points * delete_percentage;
+  int64_t        build_points = max_points * build_percentage;
+  if (delete_percentage == insert_percentage) {
+    std::cout << " random insertion and deletion" << std::endl;
+    std::vector<int> streaming_numbers;
+    for (int64_t j = 0; j < (int64_t) build_points - 1; j++) {
+      streaming_numbers.push_back(j);
+    }
+    random_shuffle(streaming_numbers.begin(), streaming_numbers.end());
 
 #pragma omp parallel for num_threads(thread_count) schedule(dynamic)
-  for (int64_t j = start; j < (int64_t) end; j++) {
-    index.insert_point(&data[(j - start) * aligned_dim], static_cast<TagT>(j));
+    for (int64_t i1 = 0; i1 < insert_points; i1++) {
+      int64_t k1 = streaming_numbers[i1];
```

Comment:
That's confuse, you are using build_points to create streaming_numbers, but use insert_points for looping, what if insert_points larger than build_points?

Reply:
My random insertion scheme is to randomly select from the complete set when there are insertions and deletions. Random sampling is achieved by shuffling the full data set and then quantitatively taking it.